android/app/src/main/java/org/smartregister/fhircore/fragment/PatientDetailsCard.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.fragment
+
+import android.content.Context
+import org.hl7.fhir.r4.model.Immunization
+import org.hl7.fhir.r4.model.Patient
+import org.hl7.fhir.r4.model.PositiveIntType
+import org.smartregister.fhircore.R
+import org.smartregister.fhircore.util.Utils
+
+/** * A wrapper class that displays a patients activity history */
+data class PatientDetailsCard(
+  val groupIndex: Int,
+  val index: Int,
+  val id: String,
+  val type: String,
+  val title: String,
+  val details: String
+)
+
+fun Patient.toDetailsCard(context: Context, index: Int = 0) =
+  PatientDetailsCard(
+    groupIndex = 0,
+    index = index,
+    id = this.id,
+    type = this.resourceType.name,
+    title = context.getString(R.string.registered_date),
+    context.getString(R.string.view_registration_details)
+  )
+
+fun Immunization.toDetailsCard(context: Context, index: Int = 0, hasNext: Boolean = false) =
+  PatientDetailsCard(
+    groupIndex = 1,
+    index = index,
+    id = this.id,
+    type = this.resourceType.name,
+    title =
+      context.getString(
+        R.string.immunization_brief_text,
+        this.vaccineCode.text,
+        (this.protocolApplied[0].doseNumber as PositiveIntType).value
+      ),
+    if (hasNext)
+      context.getString(
+        R.string.immunization_next_dose_text,
+        ((this.protocolApplied[0].doseNumber as PositiveIntType).value + 1),
+        Utils.addDays(this.occurrenceDateTimeType.toHumanDisplay(), 28)
```

Comment:
can we move to (or get) 28 from a constant?

Reply:
done